src/mixins/maskable.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Maskable
+ *
+ * @mixin
+ *
+ * Creates an input mask that is
+ * generated from a masked str
+ *
+ * Example: mask="#### #### #### ####"
+ */
+
+import {
+  isMaskDelimiter,
+  maskText,
+  unmaskText
+} from '../util/mask'
+
+export default {
+  data: () => ({
+    selection: 0,
+    preDefined: {
+      'credit-card': '#### - #### - #### - ####',
+      'date': '##/##/####',
+      'date-with-time': '##/##/#### ##:##',
+      'phone': '(###) ### - ####',
+      'social': '###-##-####',
+      'time': '##:##',
+      'time-with-seconds': '##:##:##'
+    }
+  }),
+
+  props: {
+    fillMaskBlanks: Boolean,
+    mask: {
+      type: [Object, String],
```

Comment:
Currently the support on `mask` is for `String` type.

Reply:
Good question,it's not used now but I think it can be used for creating custom rules  - like now 'A' means `/A-Z/` automaticly converted to upper case, dev could define his own rules, like `'F' => /0-9A-F/`

Reply:
This option could be very useful at least to me. I wonder if it could be used as follows.

Handled cases:

1. Predefined : String (existing)
```javascript
    mask="credit-card"
```

2. Predefined with default options : String mapped to Object
```javascript
    mask="numeral"
```

3. Predefined with overidden options : Object
```javascript
    mask="{ formatter: 'numeral', prefix: '$', precision: 4 }"
```

4. Custom mask : String (existing)
```javascript
    mask="(##) - (AA)"
```

5. Custom mask with formatter : Object
```javascript
    mask="{ formatter: function () {}, myOption1: '23', myOption2: '?' }"
```

@johnleider Is this in line with your plan?

Reply:
Well, I would have to start with something even if it's using git branch in my project. Can I attempt a PR?